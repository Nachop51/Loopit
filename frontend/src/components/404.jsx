import { Link } from "react-router-dom";

const divStyle = {
  display: "flex",
  flexDirection: "column",
  justifyContent: "center",
  alignItems: "center",
  height: "calc(100vh - 22rem)",
};

const h2Style = {
  textAlign: "center",
  fontSize: "3rem",
  fontWeight: "bold",
  color: "#fff",
};

const pStyle = {
  fontSize: "1.5rem",
  color: "#fff",
  marginBottom: "2rem",
};

const ErrorPage = () => {
  return (
    <>
      <div style={divStyle}>
        <div>
          <h2 style={h2Style}>404 Error</h2>
          <p style={pStyle}>
            The link you were looking for does not exist or has been removed.
          </p>
        </div>
        <Link to="/">
          <button className="btn btn-expand">Go back to home</button>
        </Link>
      </div>
    </>
  );
};

export default ErrorPage;
