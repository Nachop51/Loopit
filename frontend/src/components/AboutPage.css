* {
  color: #fff;
}

.container {
  width: 1200px;
  text-align: center;
  margin: 0 auto;
  background-color: rgba(255, 255, 255, 0.178);
  min-height: 100vh;
  padding: 60px 30px;
  /* align-content: center; */
  justify-content: center;
}

.about-us {
  display: flex;
  justify-content: space-between;
  width: 1000;
  height: 400px;
  margin: 0 auto;
  margin-top: 7rem;
}

.card {
  width: 200px;
  height: 100%;
  background-color: #444;
}
