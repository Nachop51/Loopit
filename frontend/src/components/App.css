* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  font-family: "Poppins", sans-serif;
}
/* ^ Utility */

body {
  /* background-image: linear-gradient(to right, #434343 0%, black 100%); */
  background-color: #111;
}

.btn-animation {
  box-shadow: 0 5px 10px rgba(0, 0, 0, 0.1);
  background-color: #09203f;
  transition: all 0.4s;
}

.btn-animation:hover {
  transform: translateY(-2px);
  box-shadow: 0 10px 20px rgba(0, 0, 0, 0.404);
}

.btn-animation:active,
.btn-animation:focus {
  outline: none;
  transform: translateY(-1px);
  box-shadow: 0 5px 10px rgba(0, 0, 0, 0.3);
}

.btn {
  @apply rounded-lg bg-white border-none w-16 text-gray-800 border-2 border-gray-800 p-2 mx-2 cursor-pointer shadow-md font-bold;
  font-size: 12px;
  @apply sm:w-20 sm:text-base;
  @apply md:w-28;
  @apply lg:w-32 lg:text-lg;
}

.btn-gray {
  @apply bg-gray-800 text-white;
}

.btn-lily {
  @apply bg-red-500 text-white;
}
