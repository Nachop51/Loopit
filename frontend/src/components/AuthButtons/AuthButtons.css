.buttons-container {
  @apply absolute top-5 right-5;
}

.button:first-child {
  @apply bg-white text-gray-800;
}

.button {
  @apply rounded-lg bg-gray-800 border-none w-32 text-white text-base border-2 border-gray-800 p-2 m-2 cursor-pointer shadow-md font-bold;
}
