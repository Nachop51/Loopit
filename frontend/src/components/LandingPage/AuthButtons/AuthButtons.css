.buttons-container {
  top: 1.5rem;
  right: 0rem;
  @apply absolute;
  @apply sm:top-5 sm:right-12;
  @apply md:top-7 md:right-12;
  @apply lg:top-12 lg:right-12;
}

.button:first-child {
  @apply bg-white text-gray-800 mb-2 sm:mb-0;
}

.button.go {
  @apply w-56 mt-12 mx-auto text-white font-bold text-lg;
}

.heading-secondary {
  @apply text-3xl font-bold text-center;
  color: var(--primary-color);
}
