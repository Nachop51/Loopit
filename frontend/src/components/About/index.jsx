import "./AboutPage.css";
import imgDani from "../../assets/Danielle.png";
import imgNacho from "../../assets/Nacho.png";
import imgVale from "../../assets/Vale.png";
import imgMateo from "../../assets/Mateo.png";
import imgSanti from "../../assets/Santi.png";
import Logo from "../Logo";
import Footer from "../Footer";
import CardAbout from "./CardAbout";
const info = [
  {
    name: "Danielle Serafim",
    rol: "Developer - UI Designer",
    img: imgDani,
    linkedIn: "https://www.linkedin.com/in/danielle-serafim-850888156/",
    instagram: "https://www.instagram.com/_danielleigyr/",
    github: "https://github.com/DaAsuncion",
  },
  {
    name: "Ignacio Peralta",
    rol: "Full Stack Developer",
    img: imgNacho,
    linkedIn: "https://www.linkedin.com/in/ignacio-peralta-dev/",
    instagram: "https://www.instagram.com/nachop51/",
    github: "https://github.com/Nachop51",
  },
  {
    name: "Valentin Repetto",
    rol: "Backend Developer",
    img: imgVale,
    linkedIn: "https://www.linkedin.com/in/valentin-repetto-6aa6711a0/",
    instagram: "https://www.instagram.com/valerepetto14/",
    github: "https://github.com/valerepetto14",
  },
  {
    name: "Mateo Arbini",
    rol: "Backend Developer / Project Manager",
    img: imgMateo,
    linkedIn: "https://www.linkedin.com/in/mateo-arbini-1493691a8/",
    instagram: "https://www.instagram.com/mateoarbini/",
    github: "https://github.com/MateoArbini",
  },
  {
    name: "Santiago Neira",
    rol: "Frontend Developer / UX",
    img: imgSanti,
    linkedIn: "https://www.linkedin.com/in/santiago-neira-4479501b7/",
    instagram: "https://www.instagram.com/santiago.neira.771/",
    github: "https://github.com/sanei1509",
  },
];
const About = () => {
  return (
    <>
      <Logo link="/" />
      <main className="about">
        <h1 className="heading-primary">Our team Project</h1>
        <div className="about-container">
          {info.map((item, i) => {
            return <CardAbout user={item} key={i} />;
          })}
        </div>
      </main>
      <Footer />
    </>
  );
};

export default About;
