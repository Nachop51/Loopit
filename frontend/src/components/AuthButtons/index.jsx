import "./AuthButtons.css";
import React, { useState } from "react";
import { connect } from "react-redux";

import ModalLogIn from "./ModalLogIn";
import ModalSignUp from "./ModalSignUp";
import { signOut } from "../../actions";

const AuthButtons = ({ signOut, isSignedIn }) => {
  const [loginIsOpen, setLoginIsOpen] = useState(false);
  const [signIsOpen, setSignIsOpen] = useState(false);

  if (isSignedIn === null) {
    return null;
  }

  if (isSignedIn) {
    return (
      <div className="buttons-container">
        <button className="button" onClick={signOut}>
          Sign Out
        </button>
      </div>
    );
  }

  return (
    <div className="buttons-container">
      <button className="button" onClick={() => setLoginIsOpen(!loginIsOpen)}>
        Log In
      </button>
      <ModalLogIn
        closeModal={() => setLoginIsOpen(false)}
        openTheOther={() => {
          setSignIsOpen(true);
        }}
        show={loginIsOpen}
      />
      <button className="button" onClick={() => setSignIsOpen(!signIsOpen)}>
        Sign Up
      </button>
      <ModalSignUp
        closeModal={() => setSignIsOpen(false)}
        openTheOther={() => {
          setLoginIsOpen(true);
        }}
        show={signIsOpen}
      />
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    isSignedIn: state.auth.isSignedIn,
  };
};

export default connect(mapStateToProps, { signOut })(AuthButtons);
