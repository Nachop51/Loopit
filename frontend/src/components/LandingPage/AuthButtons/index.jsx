import "./AuthButtons.css";
import React, { useState } from "react";
import { connect } from "react-redux";

import ModalLogIn from "./ModalLogIn";
import ModalSignUp from "./ModalSignUp";
import { signOut } from "../../../actions";

export const AuthButtons = ({ signOut, isSignedIn }) => {
  const [loginIsOpen, setLoginIsOpen] = useState(false);
  const [signIsOpen, setSignIsOpen] = useState(false);

  if (isSignedIn === null) {
    return null;
  }

  if (isSignedIn) {
    return (
      <div className="buttons-container">
        <button className="btn" onClick={signOut}>
          Sign Out
        </button>
      </div>
    );
  }

  return (
    <div className="buttons-container">
      <button
        className="btn btn-primary"
        onClick={() => setLoginIsOpen(!loginIsOpen)}
      >
        Log In
      </button>
      <ModalLogIn
        closeModal={() => setLoginIsOpen(false)}
        openTheOther={() => {
          setSignIsOpen(true);
        }}
        show={loginIsOpen}
      />
      <button className="btn" onClick={() => setSignIsOpen(!signIsOpen)}>
        Sign Up
      </button>
      <ModalSignUp
        closeModal={() => setSignIsOpen(false)}
        openTheOther={() => {
          setLoginIsOpen(true);
        }}
        show={signIsOpen}
      />
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    isSignedIn: state.auth.isSignedIn,
  };
};

export default connect(mapStateToProps, { signOut })(AuthButtons);
