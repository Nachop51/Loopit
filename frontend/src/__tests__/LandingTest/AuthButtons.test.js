import { render as testRender, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { AuthButtons } from "../../components/AuthButtons";
import { Provider } from "react-redux";
import { store } from "../../store/store";
import { BrowserRouter, Routes } from "react-router-dom";

const render = (component) =>
  testRender(
    <BrowserRouter>
      <Provider store={store}>{component}</Provider>
    </BrowserRouter>
  );

describe("Renderizado y correcto funcionamiento del flujo de autenticación", () => {
  it("Debe existir un componente AuthButtons en el dom", () => {
    const { container } = render(<AuthButtons />);

    expect(container).toMatchSnapshot();
  });

  test("Ambos botones de autenticación deben encontrarse en la página", () => {
    render(<AuthButtons />);
    // screen.debug();
    // existencia de los botones
    const loginButton = screen.getByTestId("button-login");
    expect(loginButton).toBeInTheDocument();
    const registerButton = screen.getByTestId("button-register");
    expect(registerButton).toBeInTheDocument();
    // Tipo de etiqueta
    expect(loginButton.tagName).toBe("BUTTON");
    expect(registerButton.tagName).toBe("BUTTON");
    // contenido de los botones
    expect(loginButton).toHaveTextContent("Log In");
    expect(registerButton).toHaveTextContent("Sign Up");
  });
});
