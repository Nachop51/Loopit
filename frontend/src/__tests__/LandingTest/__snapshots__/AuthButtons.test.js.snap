// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderizado y correcto funcionamiento del flujo de autenticación Debe existir un componente AuthButtons en el dom 1`] = `
<div>
  <div
    class="buttons-container"
  >
    <button
      class="button"
      data-testid="button-login"
    >
      Log In
    </button>
    <div
      class="modal "
    >
      <div
        class="modal-content"
      >
        <h2>
          Log In
        </h2>
        <h4>
          To continue to Loopit
        </h4>
        <form
          class="form"
        >
          <div>
            <label
              for="username"
            >
              Username or Email
            </label>
            <input
              autocomplete="off"
              class=""
              id="username"
              name="username"
              placeholder="email@example.com"
              value=""
            />
            <span
              class="error-message "
            >
              Username is required
            </span>
          </div>
          <div>
            <label
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="off"
              class=""
              id="password"
              name="password"
              placeholder="••••••••"
              type="password"
              value=""
            />
            <span
              class="error-message "
            >
              Password is required
            </span>
          </div>
          <span
            class="error-message "
          >
            <br />
          </span>
          <button
            class="btn"
            type="submit"
          >
            Log In
          </button>
        </form>
        <div
          class="link"
        >
          <p>
            Not registered yet? 
            <button
              class="linkTo"
            >
              Sign Up
            </button>
          </p>
        </div>
      </div>
    </div>
    <button
      class="button"
      data-testid="button-register"
    >
      Sign Up
    </button>
    <div
      class="modal "
    >
      <div
        class="modal-content"
      >
        <h2>
          Sign Up
        </h2>
        <h4>
          Create a new account
        </h4>
        <form
          class="form"
        >
          <div>
            <label
              for="fullname"
            >
              Full Name
            </label>
            <input
              autocomplete="off"
              class=""
              id="fullname"
              name="fullname"
              placeholder="John Code"
              value=""
            />
            <span
              class="error-message "
            >
              Fullname is required
            </span>
          </div>
          <div>
            <label
              for="user"
            >
              Username
            </label>
            <input
              autocomplete="off"
              class=""
              id="user"
              name="user"
              placeholder="Looper"
              value=""
            />
            <span
              class="error-message "
            >
              Username is required
            </span>
          </div>
          <div>
            <label
              for="email"
            >
              Email
            </label>
            <input
              autocomplete="off"
              class=""
              id="email"
              name="email"
              placeholder="email@example.com"
              value=""
            />
            <span
              class="error-message "
            >
              Email is required
            </span>
          </div>
          <div>
            <label
              for="pass"
            >
              Password
            </label>
            <input
              autocomplete="off"
              class=""
              id="pass"
              name="pass"
              placeholder="••••••••"
              type="password"
              value=""
            />
            <span
              class="error-message "
            >
              Password is required
            </span>
          </div>
          <div>
            <label
              for="confirmPass"
            >
              Confirm Password
            </label>
            <input
              autocomplete="off"
              class=""
              id="confirmPass"
              name="confirmPass"
              placeholder="••••••••"
              type="password"
              value=""
            />
            <span
              class="error-message "
            >
              Password confirmation is required
            </span>
          </div>
          <button
            class="btn"
            type="submit"
          >
            Sign Up
          </button>
        </form>
        <div
          class="link"
        >
          <p>
            Already have an account? 
            <button
              class="linkTo"
            >
              Log In
            </button>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
